public/js/controllers/address_controller.js

```diff
@@ -80,36 +83,189 @@ function plotGraph (processedData, otherOptions) {
+    }
+  }
+
+  _replaceHistory () {
+    // see https://github.com/turbolinks/turbolinks/issues/219. This also works:
+    // window.history.replaceState(window.history.state, this.addr, this.url.href)
+    this.turbolinks.controller.replaceHistoryWithLocationAndRestorationIdentifier(this.turbolinks.Location.wrap(this.url.href), this.turbolinks.uuid())
+    this.replaceTimer = 0
+  }
+
+  _appendHistory () {
+    // same as replaceHref, but creates a new entry in history for navigating
+    // with the browsers forward and back buttons. May still not work because of
+    // TurboLinks caching behavior, I think.
+    this.turbolinks.controller.pushHistoryWithLocationAndRestorationIdentifier(this.turbolinks.Location.wrap(this.url.href), this.turbolinks.uuid())
+    this.appendTimer = 0
+  }
+
+  replace (query) {
+    this.url.set('query', this.filteredQuery(query))
+    this.replaceHref()
+  }
+
+  to (query) {
+    this.url.set('query', this.filteredQuery(query))
+    this.toHref()
+  }
+
+  filteredQuery (query) {
+    var filtered = {}
+    Object.keys(query).forEach(function (key) {
+      var v = query[key]
+      if (typeof v === 'undefined' || v === null) return
+      filtered[key] = v
+    })
+    return filtered
+  }
+
+  update (target) {
+    // Update projects the current query parameters onto the given template.
+    return this.constructor.project(target, this.parsed)
+  }
+
+  get parsed () {
+    return this.url.query
+  }
+
+  get count () {
+    return Object.keys(this.url.query).length
+  }
+
+  get (key) {
+    if (this.url.query.hasOwnProperty(key)) {
+      return TurboQuery.parseValue(this.url.query[key])
+    }
+    return null
+  }
+
+  static parseValue (v) {
+    switch (v) {
+      case 'null':
+        return null
+      case '':
+        return null
+      case 'undefined':
+        return null
+      case 'false':
+        return false
+      case 'true':
+        return true
+    }
+    if (!isNaN(parseFloat(v)) && isFinite(v)) {
+      if (String(v).includes('.')) {
+        return parseFloat(v)
+      } else {
+        return parseInt(v)
+      }
+    }
+    return v
+  }
+
+  static project (target, source) {
+    // project fills in the properties of the given template, if they exist in
+    // the source. Extraneous source properties are not added to the template.
+    var keys = Object.keys(target)
+    var idx
+    for (idx in keys) {
+      var k = keys[idx]
+      if (source.hasOwnProperty(k)) {
+        target[k] = this.parseValue(source[k])
+      }
+    }
+    return target
+  }
+}
+
 export default class extends Controller {
   static get targets () {
     return ['options', 'addr', 'btns', 'unspent',
-      'flow', 'zoom', 'interval', 'numUnconfirmed', 'formattedTime', 'txnCount']
+      'flow', 'zoom', 'interval', 'numUnconfirmed', 'formattedTime',
+      'pagesize', 'txntype', 'txnCount', 'qron', 'qroff']
   }
 
   initialize () {
-    var _this = this
+    var controller = this
```

Comment:
Instead of setting `this` onto a variable to pass it into new scope, we can just use fat arrow functions. ( since their function bodies have lexical scoping of `this`, see  https://toddmotto.com/es6-arrow-functions-syntaxes-and-lexical-scoping/#functionality-lexical-scoping-this ) 

So by using fat arrow function in the BLOCK_RECIEVED handler below, `this` would be the controller and the alias variable not necessary.

Reply:
I mostly do the `controller` thing for semantics (which is why `_this` just wasn't working for me), but arrow functions are definitely appropriate in many places. 

Just so I know, now that you have the ability, were we planning on using [a babel plugin to translate arrow functions](https://www.npmjs.com/package/@babel/plugin-transform-arrow-functions)?

Reply:
Production builds already transpile `=>` to `function` via https://babeljs.io/docs/en/babel-preset-env being set in our webpack prod configuration.
https://github.com/decred/dcrdata/blob/master/webpack.prod.js#L27

This preset essentially says to transpile everything to ES5 that can be transpiled without polyfills.

Some things can't be transpiled, like `async`/`await`, so we need to import a polyfill for those https://github.com/decred/dcrdata/blob/master/public/index.js#L1 ( regenerator-runtime is the polyfill for async await )